docs/development/release.md

```diff
@@ -57,26 +71,40 @@ Note: This is for 1.10.3 or later.
 
 ### 6. Create a pull request against the master branch
 
-Create a PR against the `master` branch. Verify that the Travis CI build for the PR succeeds and then merge to master. Once merged, the release branch can be deleted.
+Create a PR against the feature branch. Verify that the Bamboo build for the PR succeeds and then merge to the feature branch. Once merged, the release branch can be deleted.
```

Comment:
the *feature* branch can be deleted.   

🤔   We need to make some of this consistent.